### PR TITLE
[T129] enable gathering with jetpack compose dependencies

### PR DESCRIPTION
**Pull request by idaosk**
- added necessary compose dependencies to build.gradle
- updated kotlin version 
- updated hilt version to be compatible with compose and and kotlin version

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -93,4 +97,7 @@ dependencies {
 
     // Coroutine
     implementation("org.jetbrains.kotlinx:kotlinx-coroutines-android:1.6.4")
+
+    // Jetpack Compose
+    implementation "androidx.compose.ui:ui:1.0.0"
```
Comment (amirziarati) [inline app/build.gradle:102]: its better if we also include following dependencies:

Compose Preview
Integration with ViewModels
Integration with Activities

Comment (amirziarati) [reply app/build.gradle:102]: we'd better use composeBom for easier versioning too. look [here](https://developer.android.com/jetpack/compose/setup)

Comment (idaosk) [reply app/build.gradle:102]: I actually did want to add these but after synching the project  did not rebuild. i will try again !

Comment (amirziarati) [reply app/build.gradle:107]: it probably needs a newer gradle - kotlin version !!
look into build fail error details and figure out whats the problem.

Comment (amirziarati) [reply app/build.gradle:107]: feel free to upload build fail message here so we can look into it together.

**Comment by idaosk** [comment]
the build fail error->

 Dependency 'androidx.activity:activity-ktx:1.8.0' requires libraries and applications that
      depend on it to compile against version 34 or later of the
      Android APIs.

      :app is currently compiled against android-33.

      Also, the maximum recommended compile SDK version for Android Gradle
      plugin 7.4.1 is 33.

      Recommended action: Update this project's version of the Android Gradle
      plugin to one that supports 34, then update this project to use
      compileSdkVerion of at least 34.

      Note that updating a library or application's compileSdkVersion (which
      allows newer APIs to be used) can be done separately from updating
      targetSdkVersion (which opts the app in to new runtime behavior) and
      minSdkVersion (which determines which devices the app can be installed
      on).

File: app/build.gradle (hunk 1 above)
Comment (idaosk) [reply app/build.gradle:107]: **what we have right now:**
id 'com.android.application' version '7.4.1' apply false
    id 'com.android.library' version '7.4.1' apply false
 
 **newer gradle versions:**
 
 id("com.android.application") version "8.1.0" apply false
    id("com.android.library") version "8.1.0" apply false

**Review by amirziarati — APPROVED** [review]

**Pull request merged**